src/Middleware/ParseDates.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace PartechGSS\ParseDates\Middleware;
+
+use Carbon\Carbon;
+use Carbon\Exceptions\InvalidFormatException;
+use Closure;
+
+class ParseDates
+{
+    private function parseOrNow($date)
+    {
+        try {
+            $date = Carbon::parse($date);
+        } catch (InvalidFormatException $e) {
+            $date = Carbon::now();
+        }

```

Comment:
Wouldn't something like throwing a validation error to be converted into a 422 make more sense than defaulting to `now()`?

Reply:
We could but, I don't think we really care since it's a route parameter. If your opinion is that strong we most certainly can.

Reply:
I think it would make it a lot easier to debug if it threw some sort of error rather than silently returning extra data.

Reply:
@derekrprice see a revision to comment. Still open-minded to changing, we need to address this tho. @Ashenfalke

Reply:
If backwards compatibility is an issue, I won't insist, but if it doesn't break anything and doesn't contradict other requirements, then I think the error message is more maintainable.

Reply:
Per discussion with @Ashenfalke we will make a config with an array of overrides, if there is no override & it's a bad date, then we will return a 422 saying it's a bad request.